tests/v2/validation/provisioning/rke2/etcd_only_backup_restore_test.go

```diff
@@ -0,0 +1,286 @@
+				if snapshotToBeRestored == "" {
+					snapshotToBeRestored = snapshot.Name
+				}
+				totalClusterSnapShots++
+			}
+		}
+		if totalClusterSnapShots == etcdnodeCount {
+			return true, nil
+		}
+		return false, nil
+	})
+	require.NoError(r.T(), err)
+
+	logrus.Infof("creating a workload(w2, deployment).............")
+	wloadAfterBackupLabels := map[string]string{}
+	wloadAfterBackupLabels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.deployment-%v-%v", r.ns, wloadAfterRestore)
+	containerTemplate2 := workloads.NewContainer("ngnix", "nginx", v1.PullAlways, []v1.VolumeMount{}, []v1.EnvFromSource{})
+	podTemplate2 := workloads.NewPodTemplate([]v1.Container{containerTemplate2}, []v1.Volume{}, []v1.LocalObjectReference{}, wloadAfterBackupLabels)
+	deploymentAfterBackup := workloads.NewDeploymentTemplate(wloadAfterRestore, r.ns, podTemplate2, wloadAfterBackupLabels)
+
+	deploymentResp, err = createDeployment(deploymentAfterBackup, steveclient, client, clusterID)
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), deploymentAfterBackup.Name, deploymentResp.ObjectMeta.Name)
+	logrus.Infof("%v is ready.............", deploymentAfterBackup.Name)
+
+	logrus.Infof("restoring snapshot.............")
+	require.NoError(r.T(), restoreSnapshot(client, clusterName, snapshotToBeRestored, 1, "", r.ns))
+	logrus.Infof("successfully submitted restoration request.............")
+
+	logrus.Infof("creating watch over cluster after restore.............")
+	r.watchAndWaitForCluster(kubeProvisioningClient, clusterName)
+	logrus.Infof("cluster is active again.............")
+
+	logrus.Infof("creating watch over pods.............")
+	r.watchAndWaitForPods(client, clusterID)
+	logrus.Infof("All pods are up and running.............")
+
+	logrus.Infof("fetching deployment list to validate restore.............")
+	deploymentList, err := steveclient.SteveType(workloads.DeploymentSteveType).List(nil)
+	require.NoError(r.T(), err)
+	require.Equal(r.T(), 1, len(deploymentList.Data))
+	require.Equal(r.T(), wloadBeforeRestore, deploymentList.Data[0].ObjectMeta.Name)
+	logrus.Infof(" deployment list validated successfully.............")
+
+	logrus.Infof("fetching ingresses list to validate restore.............")
+	ingressResp, err = steveclient.SteveType(ingresses.IngressSteveType).ByID(r.ns + "/" + ingressBeforeBackup.Name)
+	require.NoError(r.T(), err)
+	require.NotNil(r.T(), ingressResp)
+	logrus.Infof("ingress validated successfully.............")
+
+}
+
+func (r *RKE2EtcdOnlySnapshotRestore) watchAndWaitForCluster(kubeProvisioningClient *kubeProvisioning.Client, clusterName string) {
+	err := kwait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+		clusterResp, err := r.client.Steve.SteveType(ProvisioningSteveResouceType).ByID(r.ns + "/" + clusterName)
+		if err != nil {
+			return false, err
+		}
+		state := clusterResp.ObjectMeta.State.Name
+		return state != "active", nil
+	})
+	require.NoError(r.T(), err)
```

Comment:
I don't understand this extra wait/watch for the active state, the wait/watch below isn't sufficient to wait the cluster to be up?

Reply:
This is after a restore, or other updates to the cluster, he's making string the cluster is in an active state. Which I think can be an extension, since I can see this being used outside of just etcd back up restore testing.

Reply:
Ah I see, agreed, definitely a good candidate. In terms of being extensions, ideally, if it could break the poll once it's in the error state that might be even more useful, both for monitoring on the Jenkins side and reducing to waiting the default time out.